src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -53,40 +53,61 @@ public Assembly LoadFromPath(string fullPath)
 
         private Assembly LoadFromPathUnchecked(string fullPath)
         {
+            var assemblyIdentity = AssemblyIdentityUtils.TryGetAssemblyIdentity(fullPath);
+            return LoadFromPathUncheckedCore(fullPath, assemblyIdentity);
+        }
+
+        private Assembly LoadFromPathUncheckedCore(string fullPath, AssemblyIdentity identity)
```

Comment:
Yes, sure.
Additionally, I figured out that the new code will unnecessarily recompute assembly identity from full path for every call to LoadFromPathUnchecked(fullPath). I am going to add another cache for it.